test/models/host_test.rb

```diff
@@ -719,6 +719,19 @@ def teardown
       assert_equal taxonomies(:location2), host.location
     end
 
+    test 'comment updated from facts' do
+      host = Host.import_host('host')
+      assert host.import_facts(foreman_comment: 'new comment', lsbdistrelease: '6.7', operatingsystem: 'CentOS')
+      assert_equal 'new comment', host.comment
+    end
+
+    test 'existing comment never overwritten via facts' do
```

Comment:
I think this is wrong because there is a valid use case where users want to update their comments via facts.

Reply:
I am not following, I explicitly do not want to allow this. Let's be opinionated and avoid another setting.

Reply:
That is not what I meant. What I am saying is that *if* the fact is provided, then the user clearly wants to set the comment and it should be leading. Only if they don't provide the fact, we leave the comment untouched. Only allowing to set it once means the user needs to clear it manually if they want to update it.

> Let's be opinionated

I'm getting very sick of this as a defense. The current implementation is IMHO a wrong opinion and I'd :-1: it. I think we can avoid a setting, but then the implementation must make sense and be predictable.

Reply:
FWIW, I'm with @ekohl here.
Imho we should have these tests:

```ruby

describe 'a host' do
  context 'with the foreman_comment fact present' do
    it 'updates the host comment with the fact content' do
      ...
    end

    it 'clears the host comment when the fact is an empty string' do
      ...
    end
  end

  context 'when the foreman_comment fact is not present' do
    it 'does not touch the host comment' do
      ...
    end
  end
end

Reply:
@ekohl come on, I was just struggling to understand your concerns. You don't need to be offensive, I thought you want this to be optional. It's more clear now, thanks @timogoebel I will add those tests.

Reply:
So? ;-)

Reply:
I think this was already resolved, wasn't it? Fwiw, it also makes more sense to me to override the fact if this fact is present. Creating a custom fact is explicit desire.